src/app/components/shared/delete-modal/delete-modal.component.ts

```diff
@@ -28,16 +28,22 @@ import { ModalComponent } from "@menu/widget.component";
   `,
 })
 export class DeleteModalComponent extends withUnsubscribe() implements ModalComponent {
-  @Input() public successCallback: () => void;
-  public closeModal: (result: any) => void;
-  public dismissModal: (reason: any) => void;
+  @Input() public successCallback?: () => void;
+  @Input() public closeModal: (result: any) => void;
+  @Input() public dismissModal: (reason: any) => void;
```

Comment:
what's the difference between close and dismiss? they're synonyms to me

Reply:
Dismiss and close are inherited from NgbModals

Dismiss is used when the user presses escape to dismiss the modal without action or otherwise clicks away from it.

Close is used to close the modal with action (e.g. the user confirmed). This is typically useful for when there's some sort of input box in the modal that should be returned when submitted. But for our implementation, it's used to return if the user presses the confirm / delete action button.

Reply:
Fair enough. If that's ngb conventions document the inputs then to relay to the reader what they mean